### DISCUSSION TITLE
Is it possible to connect duckdb file from amazon s3 directly and query that?

### DISCUSSION THREAD

**Discussion by ashutosh6500** [duckdb/duckdb#8893] [Q&A]
Is it possible to connect duckdb file from amazon s3 directly and query that using duckdb.connect method or with any other method?

**Comment by Mause** [comment]
Yes, though you can only open it in read-only mode, as S3 does not support writing to bytes in files.

If your database is database is public:
```python
import duckdb

conn = duckdb.connect('s3://bucket/file.db', read_only=True)
```

Otherwise you'll need to set the s3 auth via (presently, unfortunately) environment variables: https://duckdb.org/docs/extensions/httpfs#configuration-1

**Reply by ashutosh6500** [reply]
Okay. Thanks for the input

**Reply by FlorianGD** [reply]
Hi, I cannot manage to make it work. I have defined the below environment variables, as I'd want to read and write to the DB.
```shell
AWS_REGION="us-east-1"
AWS_ACCESS_KEY_ID="***"
AWS_SECRET_ACCESS_KEY="***" 
DUCKDB_S3_ENDPOINT="***"
DUCKDB_S3_USE_SSL="false"
```
Then, in python

```python
>>> import duckdb
>>> conn = duckdb.connect("s3://bucket/file.db")
IOException: IO Error: Cannot open file "/home/***/s3://bucket/file.db": No such file or directory
```
Am I missing something? I can correctly access data on MinIO with a configuration
```python
>>> import duckdb
>>> conn = duckdb.connect()
>>> conn.execute("SET  s3_url_style='path'")
>>> conn.execute("from read_parquet('s3://bucket/folder/file.parquet') LIMIT 1").df()
  Col Value
0   A     1
```

**Reply by FlorianGD** [reply]
Maybe I misunderstood the `Otherwise` in the reply: is it to access a private database or is the inability to write inherent to S3/Minio?

**Comment by AlexisGado** [comment]
It's an old question but it's still not completely solved.
It looks like `connect` does not understand protocols other than local fs (with the Python API at least).
You can still use `ATTACH` though. But that does not work for my specific use case (I want to create a persistent db file through ffspec dir fs, and `ATTACH` can't create the file)
